src/main/resources/com/google/api/codegen/java/test.snip

```diff
@@ -0,0 +1,87 @@
+@extends "java/common.snip"
+@extends "java/initcode.snip"
+
+@snippet generate(xapiTest)
+  {@license()}
+
+  package {@xapiTest.packageName};
+
+  @join import : xapiTest.imports
+    import {@import};
+  @end
+
+  @@javax.annotation.Generated("by GAPIC")
+  public class {@xapiTest.name} {
+    private static LocalServiceHelper serviceHelper;
+    private {@xapiTest.apiClassName} api;
+    private ValueGenerator valueGenerator;
+
+    @@BeforeClass
+    public static void startStaticServer() {
+      {@xapiTest.mockServiceClassName} mockService = new {@xapiTest.mockServiceClassName}();
+      serviceHelper = new LocalServiceHelper("in-process-1", mockService);
+      serviceHelper.start();
+    }
+
+    @@AfterClass
+    public static void stopServer() {
+      serviceHelper.stop();
+    }
+
+    @@Before
+    public void setUp() throws IOException {
+      serviceHelper.reset();
+      {@xapiTest.apiSettingsClassName} settings = {@xapiTest.apiSettingsClassName}.defaultBuilder()
+          .provideChannelWith(serviceHelper.createChannel(), true)
+          .build();
+      api = {@xapiTest.apiClassName}.create(settings);
+      valueGenerator = new ValueGenerator();
+    }
+
+    @@After
+    public void tearDown() throws Exception {
+      api.close();
+    }
+
+    @join test : xapiTest.testCases
+      {@testCase(test)}
+
+    @end
+  }
+@end
+
+@private testCase(test)
+  @@Test
+  @@SuppressWarnings("all")
+  public void {@test.name}() {
+    {@initCode(test.initCode)}
+    {@methodCall(test)}
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    {@test.requestTypeName} request = ({@test.requestTypeName})requests[0];
+
+    @join assert : test.asserts
+      Assert.assertEquals(request.{@assert.actualValueIdentifier}(), {@assert.expectedValueIdentifier});
```

Comment:
It is a function name in Java but it may not be true for the other languages. I can rename it into actualValueGetter.